src/Kestrel.Core/Adapter/Internal/AdaptedPipeline.cs

```diff
@@ -63,13 +64,6 @@ public class AdaptedPipeline : IDuplexPipe
 
                     try
                     {
-                        if (result.IsCanceled)
```

Comment:
Why did we stop checking this? Aren't we still cancelling on the transport side?